src/scrapers/beinleumi.js

```diff
@@ -0,0 +1,265 @@
+  const tds = txnRow.innerTds;
+  return {
+    status: TRANSACTION_STATUS.COMPLETED,
+    date: getTransactionDate(tds, transactionsColsTypes),
+    description: getTransactionDescription(tds, transactionsColsTypes),
+    reference: getTransactionReference(tds, transactionsColsTypes),
+    debit: getTransactionDebit(tds, transactionsColsTypes),
+    credit: getTransactionCredit(tds, transactionsColsTypes),
+  };
+}
+
+async function getTransactionsColsTypeClasses(page) {
+  const typeClassesMap = [];
+  const typeClassesObjs = await pageEvalAll(page, `${TRANSACTIONS_TABLE} tbody tr:first-of-type td`, null, (tds) => {
+    return tds.map((td, index) => ({
+      colClass: td.getAttribute('class'),
+      index,
+    }));
+  });
+
+  for (const typeClassObj of typeClassesObjs) {
+    typeClassesMap[typeClassObj.colClass] = typeClassObj.index;
+  }
+  return typeClassesMap;
+}
+
+function extractTransaction(txns, txnRow, transactionsColsTypes) {
+  const txn = extractTransactionDetails(txnRow, transactionsColsTypes);
+  if (txn.date !== '') {
+    txns.push(txn);
+  }
+}
+
+async function extractTransactions(page) {
+  const txns = [];
+  const transactionsColsTypes = await getTransactionsColsTypeClasses(page);
+
+  const transactionsRows = await pageEvalAll(page, `${TRANSACTIONS_TABLE} tbody tr`, [], (trs) => {
+    return trs.map((tr) => ({
+      innerTds: Array.from(tr.getElementsByTagName('td')).map((td) => td.innerText),
+    }));
+  });
+
+  for (const txnRow of transactionsRows) {
+    extractTransaction(txns, txnRow, transactionsColsTypes);
+  }
+  return txns;
+}
+
+async function isNoTransactionInDateRangeError(page) {
+  const hasErrorInfoElement = await elementPresentOnPage(page, `.${ERROR_MESSAGE_CLASS}`);
+  if (hasErrorInfoElement) {
+    const errorText = await page.$eval(`.${ERROR_MESSAGE_CLASS}`, (errorElement) => {
+      return errorElement.innerText;
+    });
+    return errorText.trim() === NO_TRANSACTION_IN_DATE_RANGE_TEXT;
+  }
+  return false;
+}
+
+async function searchByDates(page, startDate) {
+  await clickButton(page, 'a#tabHeader4');
+  await waitUntilElementFound(page, 'div#fibi_dates');
+  await fillInput(
+    page,
+    'input#fromDate',
+    startDate.format(DATE_FORMAT),
+  );
+  await clickButton(page, `button[class*=${CLOSE_SEARCH_BY_DATES_BUTTON_CLASS}]`);
+  await clickButton(page, `input[value=${SHOW_SEARCH_BY_DATES_BUTTON_VALUE}]`);
+  await waitForNavigation(page);
+}
+
+async function getAccountNumber(page) {
+  const selectedSnifAccount = await page.$eval(ACCOUNTS_NUMBER, (option) => {
+    return option.innerText;
+  });
+
+  return selectedSnifAccount.replace('/', '_');
+}
+
+async function navigateToNextPage(page) {
+  await clickButton(page, NEXT_PAGE_LINK);
+  await waitForNavigation(page);
+}
+
+async function scrapeTransactions(page) {
+  const txns = [];
+  for (let i = 0; i < TRANSACTIONS_PAGINATION_LIMIT; i += 1) {
+    txns.push(...await extractTransactions(page));
```

Comment:
@dudiventura could you extract to a variable?
```suggestion
    const transactions = await extractTransactions(page);
    txns.push(...transactions);
```

Reply:
I agree, it will look cleaner that way.
Will do that and push on the next commit.